### PR TITLE
Xen firewall demo

### PR DESCRIPTION
**Pull request by Asiderr**

### CODE REVIEW

File: conf/layer.conf [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+# We have recipes-* directories, add to BBFILES
+BBFILES += "${LAYERDIR}/recipes-*/*/*.bb \
+            ${LAYERDIR}/recipes-*/*/*.bbappend"
+
+BBFILE_COLLECTIONS += "meta-trenchboot-demo"
+BBFILE_PATTERN_meta-trenchboot-demo = "^${LAYERDIR}/"
+BBFILE_PRIORITY_meta-trenchboot-demo = "6"
+
+LAYERDEPENDS_meta-trenchboot-demo = "core openembedded-layer"
```
Comment (macpijan) [inline conf/layer.conf]: @Asiderr This layer depends on meta-trehncboot, meta-pcengines etc. It should be reflected here

Comment (Asiderr) [reply conf/layer.conf]: @macpijan Fixed in the last commit.

File: README.md [hunk 2]
```diff
@@ -1,2 +1,28 @@
...
+
+
+# Building
+
+```
+SHELL=bash kas-docker --ssh-dir ~/ssh-keys build meta-trenchboot-demo/kas.yml
+```
+
+Change directory to build/tmp/deploy/images/pcengines-apu2 and flash the drive
+with bmap-tools:
```
Comment (pietrushnic) [inline README.md]: @Asiderr this sound like assumption that we compile on apu2, which would be unusual.

File: README.md [hunk 3]
```diff
@@ -1,2 +1,28 @@
...
+
+```
+bmaptool copy --bmap xen-dom0-image-pcengines-apu2.wic.bmap xen-dom0-image-pcengines-apu2.wic.gz /dev/<dev-dir>
```
Comment (pietrushnic) [inline README.md]: @Asiderr if you use parameters `<dev-dir>` then name should be descriptive and described in text. I have no idea what you mean by `<dev-dir>`

**Review by pietrushnic — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Asiderr) [reply README.md]: @pietrushnic I precised that in the last commit.

File: README.md (hunk 3 above)
Comment (Asiderr) [reply README.md]: @pietrushnic I added the description, with instructions on how to get that parameter.

**Comment by pietrushnic** [comment]
@Asiderr @macpijan I'm getting this:

```
WARNING: linux-yocto-4.19.72+gitAUTOINC+a7cb57afb9_2d7c98a674-r0 do_kernel_configcheck: [kernel config]: This BSP sets config options that are not offered anywhere within this kernel:

CONFIG_XEN_SCRUB_PAGES
CONFIG_XEN_MAX_DOMAIN_MEMORY
````

Is this expected?

**Review by pietrushnic — APPROVED** [review]

**Comment by pietrushnic** [comment]
Tested by booting with firmware build from [pcengines_tranchboot branch](https://github.com/pcengines/coreboot/tree/pcengines_trenchboot)

**Pull request merged**